### PR TITLE
Add support for zarith signed ints

### PR DESCRIPTION
**Pull request by gsgalloway**

### CODE REVIEW

File: zarith/zarith.go [hunk 1]
```diff
@@ -128,15 +208,91 @@ func Encode(value *big.Int) ([]byte, error) {
...
+	}
+
+	// Trim off the sign flag from the first segment
+	firstSegment := []rune(segments[0])
+	isNegative := firstSegment[0] == '1'
+	segments[0] = string(firstSegment[1:])
+
+	// Reverse the order of the segments.
+	// Source: https://github.com/golang/go/wiki/SliceTricks#reversing
+	for i := len(segments)/2 - 1; i >= 0; i-- {
```
Comment (vikstrous2) [inline zarith/zarith.go]: I prefer counting up, but it should work either way. IMO the two indices version is easier to read btw

**Comment by vikstrous2** [comment]
Re: unparam: if we don't expect that function to ever fail, it's nice to not return an error from it, so I guess I agree with the linter. I don't think there will be any reason to fail in the future either.

**Comment by gsgalloway** [comment]
> Re: unparam: if we don't expect that function to ever fail, it's nice to not return an error from it, so I guess I agree with the linter. I don't think there will be any reason to fail in the future either.

Yeah, I'd agree.The only reason I left it there was because that method is trying to conform to `encoding.BinaryMarshaler`

**Comment by vikstrous2** [comment]
Ohh in that case, let's ignore the warning

**Comment by gsgalloway** [comment]
Sorry, I'd thought your comment was referring to this line in this other PR
https://github.com/anchorageoss/tezosprotocol/pull/11/files#diff-2d786adcf2ee7bdb03ada955de5ba70dR159

Yeah no, this EncodeSigned method isn't implementing any interface. I'll take out the `error` return value

**Comment by vikstrous2** [comment]
92% diff coverage  🔥

**Pull request merged**